dev/run-tests.py

```diff
@@ -385,6 +385,10 @@ def run_sparkr_tests():
 
     if which("R"):
         run_cmd([os.path.join(SPARK_HOME, "R", "install-dev.sh")])
+        # R style check should be executed after `install-dev.sh`.
+        # Since warnings about `no visible global function definition` appear
+        # without the installation. SEE ALSO: SPARK-9121.
+        run_cmd([os.path.join(SPARK_HOME, "dev", "lint-r")])
```

Comment:
If we place the lint check in `run_sparkr_tests` then this may result in us running those lint checks after running other tests, which will make it take longer to discover R style failures in scenarios where non-R tests had to be run.  As a result, I wonder if we should place this linting check near line 488, by the existing `run_python_style_checks()` call.